test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -94,55 +96,44 @@ var _ = BeforeSuite(func() {
 	}
 	masterSSHPrivateKeyFilepath = cfg.GetSSHKeyPath()
 	longRunningApacheDeploymentName = "php-apache-long-running"
+	kubeConfig, err = GetConfig()
+	Expect(err).NotTo(HaveOccurred())
+	sshConn, err = remote.NewConnection(kubeConfig.GetServerName(), masterSSHPort, eng.ExpandedDefinition.Properties.LinuxProfile.AdminUsername, masterSSHPrivateKeyFilepath)
+	Expect(err).NotTo(HaveOccurred())
+	success := false
+	for i := 0; i < sshRetries; i++ {
+		err := util.AddToSSHKeyChain(masterSSHPrivateKeyFilepath)
```

Comment:
Also invoking `ssh-add` just once at the beginning of the test run as opposed to with each ssh execution.